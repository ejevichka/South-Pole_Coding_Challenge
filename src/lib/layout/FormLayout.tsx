import FlightForm from '~/lib/layout/Form';
import { Typography, Box } from "@mui/material"


//parrent component in terms of react composition pattern
function FormLayout() {

  return (
    <Box>
      <Typography data-testid="heading_h1" variant="h1" component="h1">
        CALCULATE YOUR FOOTPRINT
      </Typography>
      <Typography data-testid="heading_h6" variant="h6" component="h6">
        Understand your footprint and compensate for your emissions. It takes less than a minute to measure your impact.
      </Typography>
      <FlightForm />
    </Box>
  );
}

export default FormLayout;


