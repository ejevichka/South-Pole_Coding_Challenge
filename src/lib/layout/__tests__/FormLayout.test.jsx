import { render, screen } from '@testing-library/react';
import FormLayout from '~/lib/layout/FormLayout';

describe('FormLayout', () => {
  test('renders correctly', () => {
    render(<FormLayout />);
    // Assert snapshot
    expect(screen.getByTestId('heading_h1')).toMatchSnapshot();
  });

  test('displays "CALCULATE YOUR FOOTPRINT" heading', () => {
    render(<FormLayout />);
    // Assert presence of heading text 
    expect(screen.getByTestId('heading_h1')).toBeInTheDocument();
    expect(screen.getByTestId('heading_h6')).toBeInTheDocument();
  });
});
