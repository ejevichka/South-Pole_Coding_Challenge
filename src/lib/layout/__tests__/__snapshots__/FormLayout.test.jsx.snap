// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLayout renders correctly 1`] = `
<h1
  class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  data-testid="heading_h1"
>
  CALCULATE YOUR FOOTPRINT
</h1>
`;
